STYLEGUIDE.md

```diff
@@ -0,0 +1,716 @@
+Of course more fine granular checks should be applied if semantically needed.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use explicit comparison with undefined for objects */
+if(obj !== undefined) { ... }
+
+/* Don't use explicit comparison for array length (or other numbers)*/
+if(arr.length !== 0) { ... }
+
+/* Don't use explicit comparison for booleans */
+if(isFoo !== false) { ... }
+
+/* Don't use explicit comparison when checking for empty string */
+if(someString !== '') { ... }
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use truthy/falsy check for objects */
+if(obj) { ... }
+
+/* Use truthy/falsy check for array length (or other numbers)*/
+if(arr.length) { ... }
+
+/* Use truthy/falsy for booleans */
+if(!isFoo) { ... }
+
+/* Use truthy/falsy check when checking for empty string */
+if(someString) { ... }
+```
+
+## White space
+
+### Indentation: Use 2 spaces, not tabs ✓
+Use 2 spaces for indentation in all code files, unless not possible for some reason.
+Don't use tabs.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't use 4 spaces */
+arr.forEach(item => {
+····const property = item.property;
+});
+
+/* Don't use tabs */
+arr.forEach(item => {
+⇥const property = item.property;
+});
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Use 2 spaces */
+arr.forEach(item => {
+··const property = item.property;
+});
+```
+
+### Add one empty line before and after function definitions
+
+Add exactly one empty line before and after every function definition.
+
+❌ Examples of **incorrect** code:
+```ts
+/* Don't add no lines */
+function foo() {
+  ...
+}
+function bar() {
+  ...
+}
+
+/* Don't add more than one line */
+function foo() {
+  ...
+}
+
+
+function bar() {
+  ...
+}
+```
+
+✅ Examples of **correct** code:
+```ts
+/* Add one line */
+function foo() {
+  ...
+}
+
+function bar() {
+  ...
+}
+```
+
+## Comments
+
+### Document public API with TypeDoc comments
+API that is exposed publicly, should be fully documented.
+TypeDoc comments should start with `/**`.
+Functions that return something should specify the return value with `@returns`.
+Parameters should not have type annotations as those can be inherently retrieved from the TypeScript code.
+Documentation comments should have the following structure:
+1. If necessary, one line for deprecation.
+2. Description, that can extend over multiple lines.
+3. If necessary, one line per type parameter.
+4. One line per parameter.
+5. If necessary, one line for the description of the return value.
+
+There should be no empty lines in between, except if it is part of the description.
+
+### Use `@deprecated` tag for deprecation
+When deprecating public API, this is done through the TypeDoc comments.
+The first line of the comment should start with `@deprecated` followed by a note since which version this is deprecated (e. g. `Since v1.0.0.`) and a note what to use instead (or alternatively that it won't be replaced).
```

Comment:
```suggestion
### Document public API with TypeDoc comments
API that is exposed publicly, should be fully documented.
TypeDoc comments should start with `/**`.
Functions that return something should specify the return value with `@returns`.
Parameters should not have type annotations as those can be inherently retrieved from the TypeScript code.
Documentation comments should have the following structure:
1. If necessary, one line for deprecation.
2. Description, that can extend over multiple lines.
3. If necessary, one line per type parameter.
4. One line per parameter.
5. If necessary, one line for the description of the return value.

There should be no empty lines in between, except if it is part of the description.

### Use `@deprecated` tag for deprecation
When deprecating public API, this is done through the TypeDoc comments.
The first line of the comment should start with `@deprecated` followed by a note mentioning since which version this is deprecated (e. g. `Since v1.0.0.`) and a note of what to use instead (or alternatively that it won't be replaced).
```